osf/models/node.py

```diff
@@ -2430,6 +2433,29 @@ def _remove_from_associated_collections(self, auth=None, force=False):
                     force=True
                 )
 
+    def get_addon(self, name, is_deleted=False):
```

Comment:
My instinct is that, instead of having the subclasses completely override get_addon, the superclass should start with a check for the GravyValet flag and then invoke a _get_addon_from_gravyvalet or similar method that is defined on the subclass. That gets rid of the need to call super() and better delegates responsibility to the subclass